### ISSUE TITLE
NameError in userauth.py

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3154]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)| @jml reported|
|-|-|
|Trac ID|trac#3154|
|Type|defect|
|Created|2008-04-03 07:43:36Z|
|Branch|https://github.com/twisted/twisted/tree/conch-name-error-3154|

`twisted.conch.ssh.userauth.SSHUserAuthServer.ssh_USERAUTH_REQUEST` refers to `ConchError`, which is not defined. It should say `error.ConchError` instead.

I'm not sure that this code is reached in normal Conch operation. I found it with pyflakes + flymake.

No patch this time :)

<details><summary>Searchable metadata</summary>

```
trac-id__3154 3154
type__defect defect
reporter__jml jml
priority__highest highest
milestone__ 
branch__branches_conch_name_error_3154 branches/conch-name-error-3154
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1207208616000000 1207208616000000
changetime__1208735817000000 1208735817000000
version__None None
owner__ 
cc__z3p
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_1"></a>|@jml commented|
|-|-|

(In [23177]) Branching to 'conch-name-error-3154'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve commented|
|-|-|

Note that it will conflict pretty badly with [#2682](https://github.com/twisted/twisted/issues/2682).

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_3"></a>|@jml commented|
|-|-|

(In [23178]) Fix the NameError and add a test that exercises that code path.

This commit sponsored by Canonical Ltd.


refs [#3154](https://github.com/twisted/twisted/issues/3154)

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_4"></a>|@jml set owner to z3p|
|-|-|

This might not be useful, given other branches that exist.

Still, here it is.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph set owner to @jml|
|-|-|

I'd say that this should be merged, conflicts notwithstanding.  There's so little code here that the resolution can't possibly be that difficult.  Perhaps if [#2682](https://github.com/twisted/twisted/issues/2682) had taken a more incremental approach, as this ticket does, we wouldn't have this problem in the first place :).

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_6"></a>|@jml set status to `closed`|
|-|-|

(In [23362]) Fix `NameError` in userauth.py.

 * Author: jml (sponsored by Canonical Ltd)
 * Reviewers: therve, glyph
 * Fixes [#3154](https://github.com/twisted/twisted/issues/3154)

Adds a test that ensures that Conch will raise a `ConchError` in the event
of bad authentication.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
